### DISCUSSION TITLE
Cannot update MYSQL

### DISCUSSION THREAD

**Discussion by karchung0930** [leokhoa/laragon#684] [Q&A]
I follow the steps in: https://www.kreaweb.be/laragon-update-mysql/

When I try to start mysql, it fails. The error logs:

`
2024-02-09T06:09:40.066390Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-02-09T06:09:40.130679Z 0 [Warning] [MY-010097] [Server] Insecure configuration for --secure-file-priv: Current value does not restrict location of generated files. Consider setting it to a valid, non-empty path.
2024-02-09T06:09:40.130738Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
2024-02-09T06:09:40.130754Z 0 [System] [MY-010116] [Server] C:\laragon-6.0.0\bin\mysql\mysql-8.3.0-winx64\bin\mysqld (mysqld 8.3.0) starting as process 2144
2024-02-09T06:09:40.153711Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-02-09T06:09:40.382654Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
mysqld: Table 'mysql.plugin' doesn't exist
2024-02-09T06:09:40.525917Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
2024-02-09T06:09:40.526642Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.527224Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.527792Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.528410Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.528951Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.529490Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.530035Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.646355Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-02-09T06:09:40.743321Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-02-09T06:09:40.754327Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-02-09T06:09:40.754678Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-02-09T06:09:40.759798Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:40.760379Z 0 [ERROR] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-001146 - Table 'mysql.component' doesn't exist
2024-02-09T06:09:40.761229Z 0 [Warning] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-003543 - The mysql.component table is missing or has an incorrect definition.
2024-02-09T06:09:40.766429Z 0 [ERROR] [MY-010326] [Server] Fatal error: Can't open and lock privilege tables: Table 'mysql.user' doesn't exist
2024-02-09T06:09:40.766959Z 0 [ERROR] [MY-010952] [Server] The privilege system failed to initialize correctly. For complete instructions on how to upgrade MySQL to a new version please see the 'Upgrading MySQL' section from the MySQL manual.
2024-02-09T06:09:40.767942Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-02-09T06:09:42.003254Z 0 [System] [MY-010910] [Server] C:\laragon-6.0.0\bin\mysql\mysql-8.3.0-winx64\bin\mysqld: Shutdown complete (mysqld 8.3.0)  MySQL Community Server - GPL.
2024-02-09T06:09:42.003823Z 0 [System] [MY-015016] [Server] MySQL Server - end.
2024-02-09T06:09:50.074154Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-02-09T06:09:50.146331Z 0 [Warning] [MY-010097] [Server] Insecure configuration for --secure-file-priv: Current value does not restrict location of generated files. Consider setting it to a valid, non-empty path.
2024-02-09T06:09:50.146390Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
2024-02-09T06:09:50.146405Z 0 [System] [MY-010116] [Server] C:\laragon-6.0.0\bin\mysql\mysql-8.3.0-winx64\bin\mysqld (mysqld 8.3.0) starting as process 10580
2024-02-09T06:09:50.171402Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-02-09T06:09:50.401422Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
mysqld: Table 'mysql.plugin' doesn't exist
2024-02-09T06:09:50.547767Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
2024-02-09T06:09:50.548507Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.549076Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.549632Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.550233Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.550773Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.551316Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.551848Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.662802Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-02-09T06:09:50.706624Z 0 [Warning] [MY-010015] [Repl] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2024-02-09T06:09:50.717450Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-02-09T06:09:50.717801Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-02-09T06:09:50.723485Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-02-09T06:09:50.724071Z 0 [ERROR] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-001146 - Table 'mysql.component' doesn't exist
2024-02-09T06:09:50.724919Z 0 [Warning] [MY-013129] [Server] A message intended for a client cannot be sent there as no client-session is attached. Therefore, we're sending the information to the error-log instead: MY-003543 - The mysql.component table is missing or has an incorrect definition.
2024-02-09T06:09:50.732770Z 0 [ERROR] [MY-010326] [Server] Fatal error: Can't open and lock privilege tables: Table 'mysql.user' doesn't exist
2024-02-09T06:09:50.733313Z 0 [ERROR] [MY-010952] [Server] The privilege system failed to initialize correctly. For complete instructions on how to upgrade MySQL to a new version please see the 'Upgrading MySQL' section from the MySQL manual.
2024-02-09T06:09:50.734242Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-02-09T06:09:51.947797Z 0 [System] [MY-010910] [Server] C:\laragon-6.0.0\bin\mysql\mysql-8.3.0-winx64\bin\mysqld: Shutdown complete (mysqld 8.3.0)  MySQL Community Server - GPL.
2024-02-09T06:09:51.948401Z 0 [System] [MY-015016] [Server] MySQL Server - end.
`

**Comment by joseborges** [comment]
The correct installation folder should be:     **C:\laragon-6.0.0\bin\mysql\mysql-8.3.0-winx64**

Make sure you only have these "files" inside that folder:

![image](https://github.com/leokhoa/laragon/assets/7250506/f6ab8766-939f-4a65-9cb0-9f1830148d4c)


Make sure you don't have the folder:          **C:\Laragon-6.0.0\data\mysql-8**

**Reply by karchung0930** [reply]
Hi, I have the exact same 'files' as per your image. I also don't have the folder: C:\Laragon-6.0.0\data\mysql-8

What I found:
Laragon provide four ways to downlaod form Github, when I use the zip method and extract it without installing, it shows the errors as I mentioned.

When I use the laragon-wamp.exe to install it on my windows 11, it works. Any reason for this? I actually prefer the zip way because I only need few of the features such as Nginx and MySQL.

**Reply by joseborges** [reply]
Zip or EXE should be the same thing, i use the ZIP (portable) version and it works flawless.

Maybe you tried the ZIP version after having the EXE version installed, and there was some windows registry conflict.

Maybe try installing with EXE, confirm it works, uninstall with uninstaller (to remove laragon correctly from windows) and then use the portable version and see if that works.

I should mention that I'm using windows10, maybe windows11 is the reason.

**Reply by joseborges** [reply]
Did you solved it?

**Reply by karchung0930** [reply]
Hi, nope. Unfortunately it still doesn't work as I follow the steps you have mentioned. The ZIP version still fails to start the MySQL if I update through the steps given in the provided link, working great with EXE version though.

**Reply by joseborges** [reply]
Well there's nothing wrong with using rhe installed version, it works the same as portable.

**Reply by sgaddo** [reply]
Hi,
sorry @joseborges, but **what if I do have the folder C:\Laragon\data\mysql-8?**
It was created after upgrading Laragon from an older version to the 6.0, because it includes a new version of mysql (8.0.30 in my installation).

If I run Laragon using the old mysql version **mysql-5.7.24-winx64** all my websites work as before.
Instead, if I run it selecting the new mysql version **mysql-8.0.30-winx64**, my local websites won't work anymore, because they are not found.
The reason probably resides is in the **my.ini** configuration file of the last mysql version, containing this line:
`datadir= "C:/laragon/data/mysql-8"`
but all my databases are still in the **C:/laragon/data/mysql** folder.

My question now: in your opinion what if I'd change this configuration file, setting the old data dir?
The databases will be "compatible" or at least "upgradable" to the 8.0.version?
Or should I maybe move all single database folders from the data/mysql to the new data/mysql-8 folder?
I just wouldn't want to lose all my data.

Thanks for any suggestion.

**Reply by joseborges** [reply]
You can name the mysql bin directory whatever you want, as long as the subfolders are the same as the structure on the "mysql zip".

**always** make backups before doing anything.

Mysql 5 and 8, have several differences that create incompatibilities with each other.

Try to dump your mysql5 database to a SQL file and try to import it to mysql8, if it succeeded without errors, then the problem in your sites is not with the database.

Mysql5 data dir usually is DATA/MYSQL and mysql8 is DATA/MYSQL-8
as long has your mysql.ini points to the right place you shouldn't have problems.

**don't point mysql 8 to mysql 5 datadir or mysql 5 to a mysql-8 datadir** they are different.

When you say your sites don't work, what does that mean exactly?

**Reply by sgaddo** [reply]
Hi Jose,

> When you say your sites don't work, what does that mean exactly?

As I said (and sorry, I wasn't accurate enough), **the databases are not found, because they're all still in the 5.7 datadir, that is C:/laragon/data/mysql**, and not in the mysql-8, so it's perfectly normal. But that's why I was asking if maybe pointing the my.ini datadir to that old dir would kinda solve the problem.

> Try to dump your mysql5 database to a SQL file and try to import it to mysql8

**Exporting a database (using mysqldump from the 5.7) and re-importing it using version 8, worked** for one of my sites, but if you have (as I do) a lot of DBs and their size is quite big, exporting all of them could be a long process and/or lead to errors.
So, in the meanwhile I found out that it's probably possible to **migrate all old MySql 5.7 databases data to MySql 8 versions**.
Basically you need to copy them from MySQL 5.7 folder to MySQL 8.0 folder, start the MySQL 8 service and launch **mysql_upgrade** command.
I still didn't try it, but if you want I'll let you know when I will.

> don't point mysql 8 to mysql 5 datadir or mysql 5 to a mysql-8 datadir they are different.

Using the previously described method (if it works) **would allow to have and use the databases in both MySQL versions**. So no, I won't change the datadir pointing to the older, it's way better to keep both, at least for testing during the update :)
